### ISSUE TITLE
Entity Centre: fetching short collectional properties for dynamically enhanced entities leads to type casting error in EQL

### ISSUE THREAD

**Issue by 01es** [fieldenms/tg#1757]
### Description

Short collectional properties are frequently required to be displayed or used for other purposes as part of Entity Centres. Currently, if an Entity Centre configuration defines summary, which leads to dynamic enhancement of entities with calculated properties, result in type casting errors at the EQL level during instantiation of values for collectional values.

This can be addressed in 2 ways:
 1. By implementing issue #1692 -- the cleanest approach.
 2. By adjusting EQL3's container instantiator to support special handling of enhanced entity instances that get assigned to  key member properties of elements for collectional properties -- this approach is potentially  easier to implement comparing to option 1, but it is not the preferred approach.

This error can be reproduced for any Entity Centre for an entity that has a short collectional property, by adding that property to the resultset and by adding a summary to one of the properties (e.g. a count). As an example refer to `WorkActivityTypeWebUiConfig`, standalone centre configuration, property `waTypeEntities` is collectional, summary for `this` is commented out -- uncomment and run to see the error.

### Expected outcome
Ability to use (e.g. display) short collectional properties on Entity Centres with summaries.

### Actual outcome 

```Java
[qtp1395098357-37      ] ERROR ils.WebUiResourceUtils 2021-10-20 01:52:23,429 - java.lang.IllegalStateException: Can't set value [ADM] of type [class fielden.work.common.WorkActivityType$$TgEntity_bdc1de5188c24106b0c7957582aa74e0$ByteBuddy$KGKaz2xB] for property [waType] of type [class fielden.work.common.WorkActivityType] due to: java.lang.IllegalArgumentException: Can not set fielden.work.common.WorkActivityType field fielden.work.common.WaTypeEntity.waType to fielden.work.common.WorkActivityType$$TgEntity_bdc1de5188c24106b0c7957582aa74e0$ByteBuddy$KGKaz2xB
java.lang.IllegalStateException: java.lang.IllegalStateException: Can't set value [ADM] of type [class fielden.work.common.WorkActivityType$$TgEntity_bdc1de5188c24106b0c7957582aa74e0$ByteBuddy$KGKaz2xB] for property [waType] of type [class fielden.work.common.WorkActivityType] due to: java.lang.IllegalArgumentException: Can not set fielden.work.common.WorkActivityType field fielden.work.common.WaTypeEntity.waType to fielden.work.common.WorkActivityType$$TgEntity_bdc1de5188c24106b0c7957582aa74e0$ByteBuddy$KGKaz2xB
	at ua.com.fielden.platform.entity.query.EntityFetcher.getEntitiesOnPage(EntityFetcher.java:57)
	at ua.com.fielden.platform.companion.AbstractEntityReader.getEntitiesOnPage(AbstractEntityReader.java:263)
	at ua.com.fielden.platform.ioc.session.SessionInterceptor.invoke(SessionInterceptor.java:73)
	at ua.com.fielden.platform.companion.AbstractEntityReader$EntityQueryPage.<init>(AbstractEntityReader.java:351)
	at ua.com.fielden.platform.companion.AbstractEntityReader$EntityQueryPage.<init>(AbstractEntityReader.java:342)
	at ua.com.fielden.platform.companion.AbstractEntityReader.firstPage(AbstractEntityReader.java:229)
	at ua.com.fielden.platform.ioc.session.SessionInterceptor.invoke(SessionInterceptor.java:73)
	at ua.com.fielden.platform.dao.GeneratedEntityDao.firstPage(GeneratedEntityDao.java:87)
	at ua.com.fielden.platform.entity_centre.review.criteria.EntityQueryCriteria.firstPage(EntityQueryCriteria.java:749)
	at ua.com.fielden.platform.entity_centre.review.criteria.EntityQueryCriteria.run(EntityQueryCriteria.java:408)
	at ua.com.fielden.platform.web.resources.webui.CentreResourceUtils.createCriteriaMetaValuesCustomObjectWithResult(CentreResourceUtils.java:317)
	at ua.com.fielden.platform.web.resources.webui.CriteriaResource.lambda$16(CriteriaResource.java:679)
	at ua.com.fielden.platform.web.utils.WebUiResourceUtils.handleUndesiredExceptions(WebUiResourceUtils.java:70)
	at ua.com.fielden.platform.web.resources.webui.CriteriaResource.put(CriteriaResource.java:558)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.restlet.resource.ServerResource.doHandle(ServerResource.java:508)
	at org.restlet.resource.ServerResource.put(ServerResource.java:1395)
	at org.restlet.resource.ServerResource.doHandle(ServerResource.java:595)
	at org.restlet.resource.ServerResource.doNegotiatedHandle(ServerResource.java:662)
	at org.restlet.resource.ServerResource.doConditionalHandle(ServerResource.java:348)
	at org.restlet.resource.ServerResource.handle(ServerResource.java:1020)
	at ua.com.fielden.platform.web.factories.webui.CriteriaResourceFactory.handle(CriteriaResourceFactory.java:82)
	at org.restlet.routing.Filter.doHandle(Filter.java:150)
	at org.restlet.routing.Filter.handle(Filter.java:197)
	at org.restlet.routing.Router.doHandle(Router.java:422)
	at org.restlet.routing.Router.handle(Router.java:641)
	at org.restlet.routing.Filter.doHandle(Filter.java:150)
	at org.restlet.routing.Filter.handle(Filter.java:197)
	at org.restlet.routing.Filter.doHandle(Filter.java:150)
	at org.restlet.routing.Filter.handle(Filter.java:197)
	at org.restlet.routing.Router.doHandle(Router.java:422)
	at org.restlet.routing.Router.handle(Router.java:641)
	at org.restlet.routing.Filter.doHandle(Filter.java:150)
	at org.restlet.routing.Filter.handle(Filter.java:197)
	at org.restlet.routing.Filter.doHandle(Filter.java:150)
	at org.restlet.routing.Filter.handle(Filter.java:197)
	at org.restlet.routing.Filter.doHandle(Filter.java:150)
	at org.restlet.engine.application.StatusFilter.doHandle(StatusFilter.java:140)
	at org.restlet.routing.Filter.handle(Filter.java:197)
	at org.restlet.routing.Filter.doHandle(Filter.java:150)
	at org.restlet.routing.Filter.handle(Filter.java:197)
	at org.restlet.engine.CompositeHelper.handle(CompositeHelper.java:202)
	at org.restlet.engine.application.ApplicationHelper.handle(ApplicationHelper.java:77)
	at org.restlet.Application.handle(Application.java:385)
	at org.restlet.routing.Filter.doHandle(Filter.java:150)
	at org.restlet.routing.Filter.handle(Filter.java:197)
	at org.restlet.routing.Router.doHandle(Router.java:422)
	at org.restlet.routing.Router.handle(Router.java:641)
	at org.restlet.routing.Filter.doHandle(Filter.java:150)
	at org.restlet.routing.Filter.handle(Filter.java:197)
	at org.restlet.routing.Router.doHandle(Router.java:422)
	at org.restlet.routing.Router.handle(Router.java:641)
	at org.restlet.routing.Filter.doHandle(Filter.java:150)
	at org.restlet.engine.application.StatusFilter.doHandle(StatusFilter.java:140)
	at org.restlet.routing.Filter.handle(Filter.java:197)
	at org.restlet.routing.Filter.doHandle(Filter.java:150)
	at org.restlet.routing.Filter.handle(Filter.java:197)
	at org.restlet.engine.CompositeHelper.handle(CompositeHelper.java:202)
	at org.restlet.Component.handle(Component.java:408)
	at org.restlet.routing.Filter.doHandle(Filter.java:150)
	at org.restlet.routing.Filter.handle(Filter.java:197)
	at org.restlet.routing.Filter.doHandle(Filter.java:150)
	at org.restlet.routing.Filter.handle(Filter.java:197)
	at org.restlet.Server.handle(Server.java:507)
	at org.restlet.engine.connector.ServerHelper.handle(ServerHelper.java:63)
	at org.restlet.engine.adapter.HttpServerHelper.handle(HttpServerHelper.java:143)
	at org.restlet.ext.jetty.JettyServerHelper$WrappedServer.handle(JettyServerHelper.java:237)
	at org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:388)
	at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:633)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:380)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:277)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105)
	at org.eclipse.jetty.io.ChannelEndPoint$1.run(ChannelEndPoint.java:104)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:336)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:313)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:171)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:129)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:383)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:882)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1036)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.IllegalStateException: Can't set value [ADM] of type [class fielden.work.common.WorkActivityType$$TgEntity_bdc1de5188c24106b0c7957582aa74e0$ByteBuddy$KGKaz2xB] for property [waType] of type [class fielden.work.common.WorkActivityType] due to: java.lang.IllegalArgumentException: Can not set fielden.work.common.WorkActivityType field fielden.work.common.WaTypeEntity.waType to fielden.work.common.WorkActivityType$$TgEntity_bdc1de5188c24106b0c7957582aa74e0$ByteBuddy$KGKaz2xB
	at ua.com.fielden.platform.entity.query.EntityFromContainerInstantiator.setPropertyValue(EntityFromContainerInstantiator.java:113)
	at ua.com.fielden.platform.entity.query.EntityFromContainerInstantiator.instantiateFully(EntityFromContainerInstantiator.java:66)
	at ua.com.fielden.platform.entity.query.EntityFromContainerInstantiatorCache.getEntity(EntityFromContainerInstantiatorCache.java:23)
	at ua.com.fielden.platform.entity.query.EntityFromContainerInstantiator.instantiate(EntityFromContainerInstantiator.java:31)
	at ua.com.fielden.platform.entity.query.EntityFromContainerInstantiator.instantiate(EntityFromContainerInstantiator.java:90)
	at ua.com.fielden.platform.entity.query.EntityFromContainerInstantiator.instantiateFully(EntityFromContainerInstantiator.java:75)
	at ua.com.fielden.platform.entity.query.EntityFromContainerInstantiatorCache.getEntity(EntityFromContainerInstantiatorCache.java:23)
	at ua.com.fielden.platform.entity.query.EntityFromContainerInstantiator.instantiate(EntityFromContainerInstantiator.java:31)
	at ua.com.fielden.platform.entity.query.EntityFetcher.instantiateFromContainers(EntityFetcher.java:98)
	at ua.com.fielden.platform.entity.query.EntityFetcher.getEntitiesOnPage(EntityFetcher.java:48)
	... 84 more
```

**Comment by 01es** [comment]
Interestingly, this error does not manifest for User Centre, which includes a summary and short collectional property `roles`.

**Comment by 01es** [comment]
Resolved by means of issue #1692.
